### PR TITLE
Enum performance improvements to solve #458

### PR DESCRIPTION
**Pull request by GeertvanHorrik**
Still WIP.

For now:

- Added NuGet reference to NUnit 2.6.4
- Added unit tests for enum values

### CODE REVIEW

**Comment by GeertvanHorrik** [comment]
@praeclarum Can you please do a quick review of the cache? I currently store 3 "expensive" operations in the cache:

1. Is it an enum
2. If so => determine if we should store as text using reflection for the attributes
3. If so => store the values and ToString() so we have the values ready at all times. We could do this "on-demand" which could save some memory but would always require a lock (where it doesn't now since we won't modify the dictionary any longer once filled).

**Comment by GeertvanHorrik** [comment]
It was easier than I though so I already applied it. The code looks like it's doing tons more, but:

1. We need to use GetType() any way
2. We need to convert to int anyway (we either need it as value or as cache index), which is both faster than doing a ToString() on the object

We could debate the cache look up, but this should be superfast since it's simply a dictionary and could result in at least the same performance since it caches the *IsEnum* value.

File: src/SQLite.cs [hunk 1]
```diff
@@ -2014,7 +2014,63 @@ public object GetValue (object obj)
...
+#if !USE_NEW_REFLECTION_API
+            IsEnum = type.IsEnum;
+#else
+            IsEnum = type.GetTypeInfo().IsEnum;
+#endif
+
+            if (IsEnum)
+            {
+                // This is a big assumption, but for now support ints only as key, otherwise we still
+                // have to pay the price for boxing
```
Comment (praeclarum) [inline src/SQLite.cs]: I wouldn't worry about the boxing cost

File: src/SQLite.cs [hunk 2]
```diff
@@ -2014,7 +2014,63 @@ public object GetValue (object obj)
...
+                EnumValues = Enum.GetValues(type).Cast<int>().ToDictionary(x => x, x => x.ToString());
```
Comment (ea-orbital8) [inline src/SQLite.cs]: Cast<int> seems to cause InvalidCastException if enum type is not int.

**Pull request merged**